config/fcos/v1_5_exp/validate_test.go

```diff
@@ -228,3 +228,39 @@ func TestValidateExtension(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateGrubUser(t *testing.T) {
+	tests := []struct {
+		in      GrubUser
+		out     error
+		errPath path.ContextPath
+	}{
+		// valid user
+		{
+			in: GrubUser{
+				Name:         "name",
+				PasswordHash: util.StrToPtr("pkcs5-pass"),
+			},
+			out:     nil,
+			errPath: path.New("yaml", "grub"),
+		},
+		// username is not specified but password is (invalid)
+		{
+			in: GrubUser{
+				Name:         "",
+				PasswordHash: util.StrToPtr("pkcs5-pass"),
+			},
+			out:     common.ErrGrubUserNameNotSpecified,
+			errPath: path.New("yaml", "grub", "users"),
+		},
+	}
```

Comment:
Username is specified and password is missing?

Reply:
That would just be setting a superuser without a password. Is that something that we want to prohibit?

Reply:
We currently don't prohibit it, so we should test for it.  But, yeah, it doesn't seem to make sense to allow adding a superuser without a password.  It's safest to forbid it for now, since we can always relax the restriction later.